sklearn/linear_model/ridge.py

```diff
@@ -299,8 +311,9 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
         if np.atleast_1d(sample_weight).ndim > 1:
             raise ValueError("Sample weights must be 1D array or scalar")
 
-        # Sample weight can be implemented via a simple rescaling.
-        X, y = _rescale_data(X, y, sample_weight)
+        if solver == 'cholesky':
+            # Sample weight can be implemented via a simple rescaling.
+            X, y = _rescale_data(X, y, sample_weight)
```

Comment:
let's separate issues there and keep old behavior. But do add a comment to explain what you just said.
thx